src/components/ReanimatedSwipeable.tsx

```diff
@@ -266,19 +266,29 @@ const Swipeable = forwardRef<SwipeableMethods, SwipeableProps>(
     const overshootLeftProp = overshootLeft;
     const overshootRightProp = overshootRight;
 
+    const updateRightElementWidth = () => {
+      'worklet';
+      if (rightOffset.value === null) {
+        rightOffset.value = rowWidth.value;
+      }
+      rightWidth.value = Math.max(0, rowWidth.value - rightOffset.value);
+    };
+
     const calculateCurrentOffset = useCallback(() => {
       'worklet';
+      updateRightElementWidth();
       if (rowState.value === 1) {
         return leftWidth.value;
       } else if (rowState.value === -1) {
-        return -rowWidth.value - rightOffset.value;
+        return -rowWidth.value - rightOffset.value!;
```

Comment:
I'd think about changing it to use ternary operator:

```tsx
      return rowState.value === 1
        ? leftWidth.value
        : rowState.value === -1
        ? -rowWidth.value - rightOffset.value!
        : 0;
```

But it has 3 options, so it may look a bit messy. It's up to you to decide 😅

Reply:
It looks ok, done in [5e8ded8](https://github.com/software-mansion/react-native-gesture-handler/pull/3145/commits/5e8ded891497cf000ebb4a8f36ba3cfb05f7d630)